spock-spring/boot3-test/src/test/groovy/org/spockframework/boot3/WebMvcTestIntegrationSpec.groovy

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2012-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.spockframework.boot3
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+
+import org.spockframework.boot3.service.HelloWorldService
+import spock.lang.Specification
+import spock.mock.DetachedMockFactory
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.boot.test.context.TestConfiguration
+import org.springframework.context.annotation.Bean
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+
+/**
+ * Integration tests for ensuring compatibility with Spring-Boot's {@link WebMvcTest} annotation.
+ */
+//tag::include[]
+@WebMvcTest
+class WebMvcTestIntegrationSpec extends Specification {
+
+  @Autowired
+  MockMvc mvc
+
+  @Autowired
+  HelloWorldService helloWorldService
+
+  def "spring context loads for web mvc slice"() {
+    given:
+    helloWorldService.getHelloMessage() >> 'hello world'
+
+    expect: "controller is available"
+    mvc.perform(MockMvcRequestBuilders.get("/"))
+      .andExpect(status().isOk())
+      .andExpect(content().string("hello world"))
+  }
+
+  @TestConfiguration
+  static class MockConfig {
+    def detachedMockFactory = new DetachedMockFactory()
+
+    @Bean
+    HelloWorldService helloWorldService() {
+      return detachedMockFactory.Stub(HelloWorldService)
```

Comment:
Do you see a need to test also `@SpringBean` and `SpringSpy`? Or it should always work in Spring Boot 3 (assuming tests for Spring Boot 2 pass)?